src/Tactic/Cong.agda

```diff
@@ -0,0 +1,185 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Macros for making working with cong more convenient
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Level renaming (suc to lsuc; _⊔_ to lmax)
+open import Relation.Binary.Definitions
+open import Tactic.Cong.Id
+open import Data.Nat renaming (_≟_ to _≟ℕ_)
+
+module Tactic.Cong
+  (recursion-limit : ℕ)
+  (_≡_ : ∀ {ℓ} {A : Set ℓ} → A → A → Set ℓ)
+  (cong : {aℓ bℓ : Level}
+          {A : Set aℓ}
+          {B : Set bℓ}
+          (f : A → B)
+          {x y : A}
+          (p : x ≡ y)
+          → f x ≡ f y)
+  (trans : ∀ {ℓ} {A : Set ℓ} → Transitive {A = A} _≡_)
+  where
+
+open import Reflection.Term hiding (_≟_)
+open import Reflection.Abstraction
+open import Reflection.Argument renaming (map to arg-map)
+open import Reflection.Apply recursion-limit
+open import Reflection.Name using () renaming (_≟_ to _≟ⁿ_)
+open import Reflection.Pattern hiding (_≟_)
+open import Data.Unit
+open import Data.List
+open import Function.Base
+open import Relation.Nullary
+open import Data.Bool
+open import Data.Product
+open import Data.Maybe hiding (_>>=_)
+
+infixr 2 _≡⌊_⌋_
+
+private
+  extract-f : Term → Term × Bool
```

Comment:
Any chance we could have a comment explaining what this function does? As I've found at with the ring-solver, trying to debug people's macro code is deeply unpleasant without any comments :smile:

Reply:
Sure.